src/sdk/http/http-client.ts

```diff
@@ -156,7 +158,34 @@ export abstract class HttpClient<TRequest, TResponse> implements IHttpClient {
 			}
 		});
 	}
-
+	
+	async download(path, handler): Promise<any> {
```

Comment:
This should be abstract and implemented in the `RequestHttpClient` because this still relies on implementation details that should not be a concern of the base HttpClient